### PR TITLE
improve error handling

### PR DESCRIPTION
**Pull request by liamsi**
close: #297

### CODE REVIEW

File: x/payment/types/payformessage_test.go [hunk 1]
```diff
@@ -257,105 +256,6 @@ func TestSignMalleatedTxs(t *testing.T) {
 	}
 }
 
-func TestWirePayForMessage_ValidateBasic(t *testing.T) {
```
Comment (liamsi) [inline x/payment/types/payformessage_test.go:260]: Moved to x/payment/types/wirepayformessage_test.go

Test there also:
 - is slightly simpler
 - not string based
 - ensures sdk errors / abci codes

File: x/payment/types/wirepayformessage.go [hunk 2]
```diff
@@ -83,38 +81,42 @@ func (msg *MsgWirePayForMessage) ValidateBasic() error {
...
+			return ErrCommittedSquareSizeNotPowOf2.Wrapf("committed to square size: %d", commit.K)
 		}
 
 		calculatedCommit, err := CreateCommitment(commit.K, msg.GetMessageNameSpaceId(), msg.Message)
 		if err != nil {
-			return err
+			return ErrCalculateCommit.Wrap(err.Error())
 		}
 
-		if string(calculatedCommit) != string(commit.ShareCommitment) {
```
Comment (liamsi) [inline x/payment/types/wirepayformessage.go:116]: Was there ever a reason to cast this to string first?

File: x/payment/types/errors.go [hunk 3]
```diff
@@ -6,7 +6,12 @@ import (
...
-// x/payment module sentinel errors
 var (
-	ErrSample = sdkerrors.Register(ModuleName, 1100, "sample error")
+	ErrReservedNamespace              = sdkerrors.Register(ModuleName, 11110, "cannot use reserved namespace IDs")
+	ErrInvalidNamespaceLen            = sdkerrors.Register(ModuleName, 11111, "invalid namespace length")
+	ErrInvalidDataSize                = sdkerrors.Register(ModuleName, 11112, "data must be multiple of shareSize")
+	ErrDeclaredActualDataSizeMismatch = sdkerrors.Register(ModuleName, 11113, "declared data size does not match actual size")
+	ErrCommittedSquareSizeNotPowOf2   = sdkerrors.Register(ModuleName, 11114, "committed to invalid square size: must be power of two")
+	ErrCalculateCommit                = sdkerrors.Register(ModuleName, 11115, "unexpected error calculating commit for share")
+	ErrInvalidShareCommit             = sdkerrors.Register(ModuleName, 11116, "invalid commit for share")
```
Comment (liamsi) [inline x/payment/types/errors.go:16]: These could also have the same err code and could be collapsed into a single error instead. Different error codes might be better if the caller wants to handle the errors differently (e.g. add a different errors string instead as user output).

File: x/payment/types/wirepayformessage.go (hunk 2 above)
Comment (evan-forbes) [reply x/payment/types/wirepayformessage.go:116]: non other to compare it afaict

**Review by evan-forbes — APPROVED** [review]
this is a really good addition

**Review by adlerjohn — APPROVED** [review]

**Pull request merged**